(window.webpackJsonp=window.webpackJsonp||[]).push([[51],{326:function(t,r,e){"use strict";e.r(r);var n=e(1),a=Object(n.a)({},function(){this.$createElement;this._self._c;return this._m(0)},[function(){var t=this,r=t.$createElement,e=t._self._c||r;return e("div",{staticClass:"content"},[e("h2",{attrs:{id:"try-catch"}},[e("a",{staticClass:"header-anchor",attrs:{href:"#try-catch","aria-hidden":"true"}},[t._v("#")]),t._v(" try-catch")]),t._v(" "),e("p",[t._v("问题：")]),t._v(" "),e("p",[t._v("1.为何try里面放return，finally还会执行?")]),t._v(" "),e("div",{staticClass:"language- line-numbers-mode"},[e("pre",{pre:!0,attrs:{class:"language-text"}},[e("code",[t._v(" 1.不管代码异常不异常,finally都会执行的\n 2.如果try中遇到return，先跳去执行finally中的代码，然后在回来执行try中return --- ****\n 3.如果finally中有return的时候直接return出去并且覆盖掉try中return\n")])]),t._v(" "),e("div",{staticClass:"line-numbers-wrapper"},[e("span",{staticClass:"line-number"},[t._v("1")]),e("br"),e("span",{staticClass:"line-number"},[t._v("2")]),e("br"),e("span",{staticClass:"line-number"},[t._v("3")]),e("br")])]),e("p",[e("code",[t._v("疑问")])]),t._v(" "),e("pre",[e("code",[t._v("1.为什么try有return的时候，会先跳去finally中执行代码？什么机制？\n")])])])}],!1,null,null,null);r.default=a.exports}}]);